app/models/upload.rb

```diff
@@ -124,9 +136,20 @@ def diffed_headers
   end
 
   def csv_file_headers
-    csv = CSV.open(upload_file.tempfile, return_headers: true, encoding: 'ISO-8859-1', col_sep: col_sep)
+    csv = File.open(upload_file.tempfile, encoding: 'ISO-8859-1')
     skip_lines.to_i.times { csv.readline }
 
-    (csv.readline || []).select(&:present?).map { |header| header.downcase.strip }
+    first_line = csv.readline
+    set_col_sep(first_line)
+
+    first_line.split(col_sep).select(&:present?).map { |header| header.downcase.strip }
+  end
+
+  def set_col_sep(first_line)
```

Comment:
Can you we get a unit test for this new method?  If someone later adds/rearranges the order of the `Settings.csv_upload.column_separators`, it can break your intended behavior.  A unit test would help catch that breakage.

Reply:
added